action.yml

```
@@ -0,0 +1,92 @@
+author: kedro-viz
+
+branding:
+  icon: share
+  color: yellow
+
+inputs:
+  github_token:
+    description: "Set a GitHub token to authenticate deployment. This is autogenerated by the action."
+    required: false
+    default: ${{ github.token }}
+  project_path:
+    description: "Set your Kedro-project path to build the Kedro-Viz artifacts."
+    required: false
+    default: "."
+  telemetry_consent:
+    description: "Set your consent if you would like to participate in Kedro-Telemetry. Defaults to true"
+    required: false
+    default: true
+  python_version:
+    description: "Set a version of python for the deployment. Defaults to 3.11"
+    required: false
+    default: "3.11"
+  publishing_source:
+    description: "Set a publishing source for GitHub pages. Defaults to branch"
+    required: false
+    default: "branch"
+  publish_branch:
+    description: "Set a GitHub pages publish branch. Defaults to gh-pages"
+    required: false
+    default: "gh-pages"
+  commit_message:
+    description: "Set a commit message for the deployment. Defaults to your original commit message."
+    required: false
+    default: ${{ github.event.head_commit.message }}
+  user_name:
+    description: "Set git config user.name. Defaults to github-actions[bot]"
+    required: false
+    default: "github-actions[bot]"
+  user_email:
+    description: "Set git config user.email. Defaults to github-actions[bot]@users.noreply.github.com"
+    required: false
+    default: "github-actions[bot]@users.noreply.github.com"
+
+runs:
+  using: "composite"
+  steps:
+    # - name: Fetch the repository
+    #   uses: actions/checkout@v4
+    # - name: Set up Python
+    #   uses: actions/setup-python@v5
+    #   with:
+    #     python-version: ${{ inputs.python_version }}
+    #     cache: "pip"
+    - name: Install Kedro-Viz
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install "kedro-viz>=7.1.0"
+      shell: bash
+    - name: Consent to the use of Kedro-Telemetry
+      run: |
+        cd ${{ inputs.project_path }}
+        echo 'consent: ${{ inputs.telemetry_consent }}' > .telemetry
+      shell: bash
+    - name: Create build directory
+      run: |
+        cd ${{ inputs.project_path }}
+        if !(kedro viz build |& tee /dev/stderr | grep -i -q "Success!"); then
+          exit 1
+        fi
+      shell: bash
+    - name: Deploy to GitHub Pages with publish branch as source
+      if: ${{ inputs.publishing_source == 'branch' }}
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ inputs.github_token }}
+        publish_branch: ${{ inputs.publish_branch }}
+        publish_dir: '${{ inputs.project_path }}/build'
+        user_name: ${{ inputs.user_name }}
+        user_email: ${{ inputs.user_email }}
+        commit_message: ${{ inputs.commit_message }}
+        force_orphan: true
+    - name: Upload Pages artifact
```

```suggestion
    - name: Upload GitHub Pages artifact
```
For consistency with other places